.github/workflows/cross_version_compat.yml

```diff
@@ -0,0 +1,189 @@
+name: Cluster-Client version compatibility 
+
+on:
+  push:
+    branches: [master]
+  workflow_dispatch:
+jobs:
+  cross_compat:
+    timeout-minutes: 20
+    name: cluster(${{ matrix.cluster_type }}):${{ matrix.cluster_version }} x CLI:${{ matrix.cli_version }} 
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        rust: [stable]
+        cluster_type: [local] # fyi: k8 tests are unstable in github-runner
+        # Strategy: latest, stable, stable-1 release
+        cli_version: [latest, stable, 0.7.3]
+        cluster_version: [latest, stable, 0.7.3]
+        exclude:
+          - cli_version: latest
+            cluster_version: stable
+          - cli_version: latest
+            cluster_version: 0.7.3 
+          - cli_version: stable
+            cluster_version: 0.7.3 
```

Comment:
I think this is going to become a bottleneck for this test. I think we need to come up with a programmatic way of finding the "stable - 1" version. One potential option for this may be to use the `gh` CLI:

```
$ gh release list -R infinyon/fluvio
dev             Pre-release  (dev)             about 1 hour ago
v0.8.0          Latest       (v0.8.0)          about 3 hours ago
v0.7.3                       (v0.7.3)          about 24 days ago
v0.7.2                       (v0.7.2)          about 1 month ago
v0.7.2-alpha.1  Pre-release  (v0.7.2-alpha.1)  about 1 month ago
v0.7.1                       (v0.7.1)          about 1 month ago
v0.7.1-beta.0   Pre-release  (v0.7.1-beta.0)   about 1 month ago
v0.7.1-alpha.0  Pre-release  (v0.7.1-alpha.0)  about 1 month ago
v0.7.0                       (v0.7.0)          about 2 months ago
v0.7.0-beta.1   Pre-release  (v0.7.0-beta.1)   about 2 months ago
v0.7.0-alpha.5  Pre-release  (v0.7.0-alpha.5)  about 2 months ago
v0.7.0-alpha.4  Pre-release  (v0.7.0-alpha.4)  about 2 months ago
v0.7.0-alpha.3  Pre-release  (v0.7.0-alpha.3)  about 2 months ago
v0.6.2-alpha.2  Pre-release  (v0.6.2-alpha.2)  about 2 months ago
v0.7.0-alpha.2  Pre-release  (v0.7.0-alpha.2)  about 2 months ago
v0.7.0-alpha.1  Pre-release  (v0.7.0-alpha.1)  about 2 months ago
v0.6.2-alpha.1  Pre-release  (v0.6.2-alpha.1)  about 2 months ago
v0.6.1          Pre-release  (v0.6.1)          about 3 months ago
v0.6.0                       (v0.6.0)          about 3 months ago
v0.6.0-rc.5     Pre-release  (v0.6.0-rc.5)     about 3 months ago
v0.6.0-rc.4     Pre-release  (v0.6.0-rc.4)     about 3 months ago
v0.6.0-rc.3     Pre-release  (v0.6.0-rc.3)     about 3 months ago
v0.6.0-rc.2     Pre-release  (v0.6.0-rc.2)     about 3 months ago
v0.6.0-rc.1     Pre-release  (v0.6.0-rc.1)     about 3 months ago
v0.6.0-beta.2   Pre-release  (v0.6.0-beta.2)   about 3 months ago
v0.6.0-beta.1   Pre-release  (v0.6.0-beta.1)   about 4 months ago
v0.6.0-alpha.8  Pre-release  (v0.6.0-alpha.8)  about 4 months ago
v0.6.0-alpha.7  Pre-release  (v0.6.0-alpha.7)  about 4 months ago
v0.6.0-alpha.6  Pre-release  (v0.6.0-alpha.6)  about 4 months ago
v0.6.0-alpha.5  Pre-release  (v0.6.0-alpha.5)  about 5 months ago
```

This output seems to be optimized for human-readability, but you can also find this kind of output via the API:

```
❯ gh api repos/infinyon/fluvio/releases | jq '.[].tag_name'
"dev"
"v0.8.0"
"v0.7.3"
"v0.7.2"
"v0.7.2-alpha.1"
"v0.7.1"
"v0.7.1-beta.0"
"v0.7.1-alpha.0"
"v0.7.0"
"v0.7.0-beta.1"
"v0.7.0-alpha.5"
"v0.7.0-alpha.4"
"v0.7.0-alpha.3"
"v0.6.2-alpha.2"
"v0.7.0-alpha.2"
"v0.7.0-alpha.1"
"v0.6.2-alpha.1"
"v0.6.1"
"v0.6.0"
"v0.6.0-rc.5"
"v0.6.0-rc.4"
"v0.6.0-rc.3"
"v0.6.0-rc.2"
"v0.6.0-rc.1"
"v0.6.0-beta.2"
"v0.6.0-beta.1"
"v0.6.0-alpha.8"
"v0.6.0-alpha.7"
"v0.6.0-alpha.6"
"v0.6.0-alpha.5"
```

There is more info available depending on what `jq` filter you use